src/main/java/org/sonatype/gradle/plugins/scan/nexus/iq/index/NexusIqIndexTask.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonatype.gradle.plugins.scan.nexus.iq.index;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import com.sonatype.insight.scan.module.model.Module;
+import com.sonatype.insight.scan.module.model.io.ModuleIoManager;
+
+import org.sonatype.gradle.plugins.scan.common.DependenciesFinder;
+
+import org.apache.commons.lang3.StringUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.internal.impldep.com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NexusIqIndexTask
+    extends DefaultTask
+{
+  private final Logger log = LoggerFactory.getLogger(NexusIqIndexTask.class);
+
+  private final NexusIqPluginIndexExtension extension;
+
+  private DependenciesFinder dependenciesFinder;
+
+  private ModuleIoManager moduleIoManager;
+
+  public NexusIqIndexTask() {
+    extension = getProject().getExtensions().getByType(NexusIqPluginIndexExtension.class);
+    dependenciesFinder = new DependenciesFinder();
+    moduleIoManager = new ModuleIoManager(log);
+  }
+
+  @TaskAction
+  public void saveModule() {
+    try {
+      List<Module> modules =
+          dependenciesFinder.findModules(getProject(), extension.isAllConfigurations(), extension.getModulesExcluded());
+      List<File> files = new ArrayList<>(modules.size());
+
+      for (Module module : modules) {
+        File file = Paths.get(module.getPathname(), "build", "sonatype-clm", "module.xml").toFile();
+        moduleIoManager.writeModule(file, module);
```

Comment:
it does not let me add a comment in the line above, but Just wanted to say that `sonatype-clm` and `module.xml` are repeated in some places maybe  nice idea would be to have constants for those values

Reply:
> it does not let me add a comment in the line above, but Just wanted to say that `sonatype-clm` and `module.xml` are repeated in some places maybe nice idea would be to have constants for those values

Done: 6728e6f38ec379945b7790d2f4e2cd2e591c13f9, 2b3e9da1aa23a457fb8d4d8b3ff60e592568a631